test/capability-specs.js

```diff
@@ -0,0 +1,219 @@
+    } catch (e) {
+      e.should.be.instanceof(errors.SessionNotCreatedError);
+      e.message.should.contain('platformName');
+      e.message.should.contain('deviceName');
+      return;
+    }
+
+    should.fail('error should have been thrown');
+  });
+
+  it('should require platformName', async () => {
+    try {
+      await d.createSession({'platformName': 'iOS'});
+    } catch (e) {
+      e.should.be.instanceof(errors.SessionNotCreatedError);
+      e.message.should.contain('deviceName');
+      return;
+    }
+
+    should.fail('error should have been thrown');
+  });
+
+  it('should require deviceName', async () => {
+    try {
+      await d.createSession({'deviceName': 'Delorean'});
+    } catch (e) {
+      e.should.be.instanceof(errors.SessionNotCreatedError);
+      e.message.should.contain('platformName');
+      return;
+    }
+
+    should.fail('error should have been thrown');
+  });
+
+  it('should not care about cap order', async () => {
+
+    await d.createSession({
+      deviceName: 'Delorean',
+      platformName: 'iOS'
+    });
+
+  });
+
+  it('should check required caps which are added to driver', async () => {
+    d.desiredCapConstraints = {
+      necessary: {
+        presence: true
+      },
+      proper: {
+        presence: true,
+        isString: true,
+        inclusion: ['Delorean', 'Reventon']
+      }
+    };
+
+    try {
+      await d.createSession({
+        'platformName': 'iOS',
+        'deviceName': 'Delorean'
+      });
+    } catch (e) {
+      e.should.be.instanceof(errors.SessionNotCreatedError);
+      e.message.should.contain('necessary');
+      e.message.should.contain('proper');
+      return;
+    }
+
+    should.fail('error should have been thrown');
+  });
+
+  it('should check added required caps in addition to base', async () => {
+    d.desiredCapConstraints = {
+      necessary: {
+        presence: true
+      },
+      proper: {
+        presence: true,
+        isString: true,
+        inclusion: ['Delorean', 'Reventon']
+      }
+    };
+
+    try {
+      await d.createSession({
+        necessary: 'yup',
+        proper: 'yup, your highness'
+      });
+    } catch (e) {
+      e.should.be.instanceof(errors.SessionNotCreatedError);
+      e.message.should.contain('platformName');
+      e.message.should.contain('deviceName');
+      return;
+    }
+
+    should.fail('error should have been thrown');
+  });
+
+  it('should accept extra capabilities', async () => {
+    await d.createSession({
+      'platformName': 'iOS',
+      'deviceName': 'Delorean',
+      'extra': 'cheese',
+      'hold the': 'sauce'
+    });
+  });
+
+  it('should log the use of extra caps', async function () {
+    this.timeout(500);
+
+    let spy = sinon.spy(logger, 'warn');
+
+    await d.createSession({
+      'platformName': 'iOS',
+      'deviceName': 'Delorean',
+      'extra': 'cheese',
+      'hold the': 'sauce'
+    });
+
+    spy.callCount.should.be.above(0);
```

Comment:
Same for all test, you can just do:
`logger.warn.callCount.should.be.above(0);`